lib/rules/block-spacing.js

```diff
@@ -15,7 +15,8 @@ var util = require("../ast-utils");
 module.exports = function(context) {
     var always = (context.options[0] !== "never"),
         message = always ? "Requires a space" : "Unexpected space(s)",
-        sourceCode = context.getSourceCode();
+        sourceCode = context.getSourceCode(),
+        tokensAndComments = sourceCode.tokensAndComments;
```

Comment:
Could we use TokenStore for `sourceCode.tokensAndComments`?
`context.getTokenAfter` and `context.getTokenBefore` are `O(1)` operation but `Array.prototype.indexOf` is `O(n)` operation.
And this `n` might be huge, and this operation is inside of other loop.

https://github.com/eslint/eslint/blob/master/lib/token-store.js